sql/sql_acl.cc

```diff
@@ -11166,6 +11167,24 @@ acl_authenticate(THD *thd, uint com_change_user_pkt_len)
       general_log_print(thd, command, (char*) "%s@%s on %s",
                         mpvio.auth_info.user_name, mpvio.auth_info.host_or_ip,
                         mpvio.db.str ? mpvio.db.str : (char*) "");
+
+    if (log_legacy_user && opt_legacy_user_name_pattern)
+    {
+      std::regex pattern(opt_legacy_user_name_pattern);
```

Comment:
Creating the std::regex with an invalid pattern (such as having an opening bracket but no closing bracket) will throw an exception which will cause the server to crash.  I've added a class that will handle this case for you.  See `Regex_list_handler` (sql/handler.*).  It supports a list of regex patterns and returns an error if you try to set a bad pattern and even returns a list of the bad patterns so you can output them in the log and then continue.  You can see the use cases by searching for gap_lock_exceptions (sql/handler.cc) and rdb_collation_exceptions (storage/rocksdb/ha_rocksdb.cc).

Another benefit is that you can create the Regex_list_handler once, and then only update the pattern if `opt_legacy_user_name_pattern` changes, getting better performance than the current implementation that creates a new std::regex with every user login.